### PR TITLE
add document describing setup for a new client

### PR DESCRIPTION
**Pull request by pkarman**
Trello: https://trello.com/c/i4Koni9u

The example code is based on the `Test::ClientRequest` code in `spec/`. NOTE this example code hasn't been tested. I would expect that anyone adding a new client would amend this document based on real-world experience. This is intended as a starting place.

### CODE REVIEW

File: doc/client_setup.md [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+a data model. Existing examples include **GSA18F** and **NCR**.
+
+Within a *client* users may be assigned different roles. See the [Authorization](roles.md)
+document for details on roles.
+
+A *client* namespace is assigned to each User via a `client_slug` model value. The
+convention is that the `client_slug` is a downcased string matching the client model
+namespace. For example `ncr` for **NCR**.
+
+A User without an assigned `client_slug` may not create, edit, view or search Proposals,
```
Comment (jessieay) [inline doc/client_setup.md]: this is not *technically* true at the moment. When my client slug is set to `gsa18f`, I can still create an NCR work order if I know the URL for doing so. But the link for creating one is not surfaced in the header.

Comment (jessieay) [inline doc/client_setup.md]: same goes for users without a client slug at all. They won't see a link to create a work order or procurement, but it is possible to do so.

File: doc/client_setup.md [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+called **SpendingRequest**:
+
+```ruby
+class AddFooSpendingRequests < ActiveRecord::Migration
+  def change
+    create_table :foo_spending_requests do |t|
+      t.decimal  :amount
+      t.string   :project_title
+      t.datetime :created_at
+      t.datetime :updated_at
```
Comment (jessieay) [inline doc/client_setup.md]: super small nit but these two columns can be and are usually represented via `t.timestamps null: false`

File: doc/client_setup.md [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+
+```ruby
+FactoryGirl.define do
+  factory :foo_spending_request, class: Foo::SpendingRequest do
+    amount 123
+    project_title "I am a test request"
+    association :proposal, client_slug: "foo"
+
+    trait :with_approvers do
+      association :proposal, :with_serial_approvers, client_slug: "foo"
```
Comment (jessieay) [inline doc/client_setup.md]: should we get into describing approver flows?

**Comment by jessieay** [comment]
This document is very clear and helpful.

One thing I think we should add is some information about how to deal with approvers. Who receives emails / approves requests is an important part of any purchase request.

File: doc/client_setup.md (hunk 1 above)
Comment (pkarman) [inline doc/client_setup.md]: @jessieay you might be seeing something different than "normal" users because you are an admin. These tests demonstrate that "normal" users are limited based on `client_slug`.

https://github.com/18F/C2/blob/master/spec/features/client_slug_authz_spec.rb

File: doc/client_setup.md (hunk 2 above)
Comment (pkarman) [inline doc/client_setup.md]: agreed. fixed in 7623706

File: doc/client_setup.md (hunk 3 above)
Comment (pkarman) [inline doc/client_setup.md]: Maybe? Probably? That seems out of scope for this doc though. You can set up a new client w/o needing to know much about approver flows, though they are important.

File: doc/client_setup.md (hunk 1 above)
Comment (jessieay) [inline doc/client_setup.md]: oh ok, good to know

**Comment by jessieay** [comment]
Perhaps this should get a set of @phirefly eyes before merging?

**Comment by phirefly** [comment]
This looks great! TY
:shipit:
